kidviz/views.py

```diff
@@ -606,11 +549,35 @@ def get_context_data(self, **kwargs):
             'course_id': course_id,
             'filtering_form': date_filtering_form,
             'selected_chart': self.selected_chart(),
-            'star_matrix_vertical': star_matrix_vertical,
-            'star_matrix_by_class': star_matrix_by_class
+            'star_matrix_vertical': Observation.get_vertical_stars(star_matrix),
+            'star_matrix_by_class': star_matrix_by_class,
+            'star_chart_4': Observation.create_star_chart_4(
+                time_observations, all_students, constructs),
+            'time_observations_count': time_observations.count()
         })
         return data
 
+    def _init_filter_form(self, GET_DATA):
+        return DateFilteringForm(GET_DATA, initial={
+            'course': GET_DATA.get('course', None),
+            'date_from': GET_DATA.get('date_from', None),
+            'date_to': GET_DATA.get('date_to', None),
+            'constructs': GET_DATA.get('constructs', None),
+            'tags': GET_DATA.get('tags', None),
+            'time_window': GET_DATA.get('time_window', None)
+        })
+
+    def _get_time_observations(self, observations, time_window):
+        if time_window:
+            if time_window == 3:
+                time_window = 4
```

Comment:
What is this code? Why do you increment `time_window` if it is equal to 3? Please use constant from model or create mapping function which returns number of weeks based on time window choice.

Reply:
You can even move this function to model as static fuction which imo is a better place for such logic.